### PR TITLE
File Server Connector Bug Fixes & Improvements

### PR DESCRIPTION
**Pull request by dilini-muthumala**
- Fixing https://github.com/ballerinalang/ballerina/issues/1949 by adding a warn log.
- With reference to https://github.com/ballerinalang/ballerina/issues/1947, adding `deleteIfNotAcknowledged` parameter.
- Fixing https://github.com/ballerinalang/ballerina/issues/2099

### CODE REVIEW

**Review by Kishanthan — CHANGES_REQUESTED** [review]
Please send this PR to https://github.com/wso2/carbon-transports/tree/server-connector-bug-fixes branch

File: file/org.wso2.carbon.transport.file/src/main/java/org/wso2/carbon/transport/file/connector/server/FileConsumer.java [hunk 1]
```diff
@@ -197,6 +205,16 @@ private void setupParams() throws ServerConnectorException {
                         timeOutInterval + " milliseconds", e);
             }
         }
+        String strDeleteIfNotAck = fileProperties.get(Constants.FILE_DELETE_IF_NOT_ACKNOWLEDGED);
+        if (strDeleteIfNotAck != null) {
+            try {
+                deleteIfNotAck = Boolean.parseBoolean(strDeleteIfNotAck);
+            } catch (RuntimeException re) {
```
Comment (Kishanthan) [inline file/org.wso2.carbon.transport.file/src/main/java/org/wso2/carbon/transport/file/connector/server/FileConsumer.java]: Why catching Runtime Exception here?

File: file/org.wso2.carbon.transport.file/src/main/java/org/wso2/carbon/transport/file/connector/server/FileConsumer.java [hunk 2]
```diff
@@ -232,42 +250,51 @@ private void directoryHandler(FileObject[] children) throws FileServerConnectorE
...
+                        log.warn(Constants.FILE_SORT_ORDER + " parameter should be either " +
+                                "\"true\" or \"false\". Found: " + strSortOrder +
+                                ". Assigning default value \"true\".", re);
+                    }
+                }
+                if (log.isDebugEnabled()) {
+                    log.debug("Sorting the files by : " + strSortOrder + ". (" +
+                            bSortOrderAscending + ")");
+                }
+                if (strSortParam.equals(Constants.FILE_SORT_VALUE_NAME) && bSortOrderAscending) {
```
Comment (Kishanthan) [inline file/org.wso2.carbon.transport.file/src/main/java/org/wso2/carbon/transport/file/connector/server/FileConsumer.java]: Can we use switch case for all these ?

**Review by Kishanthan — CHANGES_REQUESTED** [review]

File: file/org.wso2.carbon.transport.file/src/main/java/org/wso2/carbon/transport/file/connector/server/FileConsumer.java (hunk 1 above)
Comment (dilini-muthumala) [reply file/org.wso2.carbon.transport.file/src/main/java/org/wso2/carbon/transport/file/connector/server/FileConsumer.java]: Removed it in https://github.com/wso2/carbon-transports/pull/228/commits/bd704ce1531f38c52ffc2b20070992a82323887d. I thought `Boolean.parseBoolean()` would throw a `RuntimeException` if the argument given is neither `true` nor `false`. I just looked at the implementation and it returns `false` for any argument other than `true` :(

File: file/org.wso2.carbon.transport.file/src/main/java/org/wso2/carbon/transport/file/connector/server/FileConsumer.java (hunk 2 above)
Comment (dilini-muthumala) [reply file/org.wso2.carbon.transport.file/src/main/java/org/wso2/carbon/transport/file/connector/server/FileConsumer.java]: +1. Done in https://github.com/wso2/carbon-transports/pull/228/commits/bd704ce1531f38c52ffc2b20070992a82323887d

**Review by Kishanthan — APPROVED** [review]

**Pull request merged**